frontend/packages/dev-console/src/components/import/git/GitSection.tsx

```diff
@@ -230,16 +230,24 @@ const GitSection: React.FC<GitSectionProps> = ({
         return;
       }
       const detectedGitType = detectGitType(url);
-      const gitType = values.git.showGitType ? values.git.type : detectedGitType;
+      const isUnsureDetectedGitType = detectedGitType === GitTypes.unsure;
       const gitRepoName = formType !== 'sample' && detectGitRepoName(url);
 
       // Updated detectedType only
       if (detectedGitType !== values.git.detectedType) {
-        setFieldValue('git.detectedType', gitType);
+        setFieldValue('git.detectedType', detectedGitType);
       }
-      if (detectedGitType === GitTypes.unsure && !values.git.showGitType) {
+      if (isUnsureDetectedGitType && !values.git.showGitType) {
         setFieldValue('git.showGitType', true);
       }
+
+      if (!isUnsureDetectedGitType && values.git.showGitType) {
+        setFieldValue('git.showGitType', false);
+      }
+
+      const gitType =
+        isUnsureDetectedGitType && values.git.showGitType ? values.git.type : detectedGitType;
+
       if (gitType !== values.git.type) {
         setFieldValue('git.type', gitType);
       }
```

Comment:
```suggestion
      if (gitType !== values.git.type) {
        setFieldTouched('git.type', false, false);
        setFieldValue('git.type', gitType, true);
      }
```

With this change:

![fixed](https://user-images.githubusercontent.com/139310/148658954-94ea1c67-74e4-46f7-8db9-d5eebdca7f6c.gif)

Reply:
Thanks for the feedback, I updated accordingly :)